src/tdycore.c

```diff
@@ -559,6 +563,10 @@ PetscErrorCode TDySetFromOptions(TDy tdy) {
   ierr = PetscOptionsEnd(); CHKERRQ(ierr);
   tdy->setupflags |= TDyOptionsSet;
 
+  if (tdy->init_from_file && tdy->init_with_random_field) {
+    SETERRQ(PETSC_COMM_WORLD,PETSC_ERR_USER,"Both -tdy_init_from_file and -tdy_init_with_random_field cannot be specified");
```

Comment:
Does `SETERRQ` automatically raise an error, or do we need to call `CHKERRQ` here to get the error message and halt the program?

Reply:
No need to call `CHKERRQ` see https://www.mcs.anl.gov/petsc/petsc-current/docs/manualpages/Sys/SETERRQ.html